tool/src/js/gen.rs

```diff
@@ -66,15 +73,37 @@ impl<'ctx, 'tcx> TyGenContext<'ctx, 'tcx> {
     /// A wrapper for `borrow_mut`ably inserting new imports.
     ///
     /// I do this to avoid borrow checking madness.
-    pub(super) fn add_import(&self, import_str: String) {
-        self.imports.borrow_mut().insert(import_str);
+    pub(super) fn add_import(
+        &self,
+        import_str: Cow<'tcx, str>,
+        import_file: Option<Cow<'tcx, str>>,
+        usage: ImportUsage,
+    ) {
+        self.imports.borrow_mut().insert(ImportInfo {
```

Comment:
thought: in the Both case should we insert twice instead? Otherwise this will lead to duplicate imports if you push a Both and a Typescript, yes?